### PR TITLE
SLLS-37 Rules telemetry

### PR DESCRIPTION
**Pull request by henryju**
This is a draft. Tests are missing!

### CODE REVIEW

File: src/test/java/org/sonarsource/sonarlint/ls/CommandManagerTests.java [hunk 1]
```diff
@@ -25,6 +25,8 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+
+import org.checkerframework.checker.nullness.Opt;
```
Comment (jblievremont) [inline src/test/java/org/sonarsource/sonarlint/ls/CommandManagerTests.java:29]: Dubious import

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintTelemetry.java [hunk 2]
```diff
@@ -149,32 +158,38 @@ public void analysisDoneOnSingleLanguage(Language language, int analysisTimeMs)
     }
   }
 
+  public void addReportedRules(Set<String> ruleKeys) {
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/SonarLintTelemetry.java:161]: I'd like to see a test for this.

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻 see my few comments

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintTelemetry.java (hunk 2 above)
Comment (kirill-knize-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/SonarLintTelemetry.java:161]: Done.

File: src/test/java/org/sonarsource/sonarlint/ls/CommandManagerTests.java (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply src/test/java/org/sonarsource/sonarlint/ls/CommandManagerTests.java:29]: Removed.

**Pull request merged**